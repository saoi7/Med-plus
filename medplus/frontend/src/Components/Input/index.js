import React from 'react';

function Input(props) {
   return (
        <div className="add-med-input font-medium">
            <label className="left-padding">
                {props.labelText + ": "}
            </label>
            <input type={props.type}
                   name={props.name}
                   value={props.value}
                   onChange={props.onChange}
                   className="hidden-text-box left-padding font-small"
                   required={props.required}
            />
        </div>
    );
}

export default Input;
