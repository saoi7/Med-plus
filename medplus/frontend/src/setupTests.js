// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { unmountComponentAtNode } from "react-dom";

let test_dom_container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  test_dom_container = document.createElement("div");
  document.body.appendChild(test_dom_container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(test_dom_container);
  test_dom_container.remove();
  test_dom_container = null;
});

export { test_dom_container };
