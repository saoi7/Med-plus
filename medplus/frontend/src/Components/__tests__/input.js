import Input from '../Input';
import {test_dom_container} from '../../setupTests';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

test("check no unintended changes have been made to rendered component", () => {
    const component= renderer.create(
        <Input />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const conponent_with_label = renderer.create(
        <Input labelText="acbdefg" />
    );
});

test("check html elements of Input component", () => {
    act(() => {
        render(<Input />, test_dom_container);
    });

    const ELEMENT = test_dom_container.childNodes;
    expect(ELEMENT[0].tagName).toBe('DIV');

    const CHILDREN = ELEMENT[0].childNodes;
    expect(CHILDREN.length).toBe(2);

    const LABEL_ELEMENT = CHILDREN[0];
    const INPUT_ELEMENT = CHILDREN[1];
    expect(LABEL_ELEMENT.tagName).toBe('LABEL');
    expect(INPUT_ELEMENT.tagName).toBe('INPUT');
});

test("check text content of label for Input component", () => {
    const LABEL_TEXT_PROP = 'abcdefg';

    act(() => {
        render(<Input labelText={LABEL_TEXT_PROP} />, test_dom_container);
    });

    const ELEMENT = test_dom_container.childNodes;
    const CHILDREN = ELEMENT[0].childNodes;
    const LABEL_ELEMENT = CHILDREN[0];

    const RENDERED_LABEL_TEXT = LABEL_ELEMENT.textContent;
    expect(RENDERED_LABEL_TEXT).toBe(LABEL_TEXT_PROP + ': ');
});

test("check input attributes are set correctly for Input component", () => {
    const TYPE_PROP = 'text';
    const NAME_PROP = 'abc';
    const VALUE_PROP = 'def';
    const REQUIRED_PROP = true;
    const DUMMY_ON_CHANGE = jest.fn();

    act(() => {
        render(
            <Input onChange={DUMMY_ON_CHANGE} type={TYPE_PROP} name={NAME_PROP} value={VALUE_PROP} required={REQUIRED_PROP} />,
            test_dom_container
        );
    });

    const ELEMENT = test_dom_container.childNodes;
    const CHILDREN = ELEMENT[0].childNodes;
    const INPUT_ELEMENT = CHILDREN[1];

    const RENDERED_INPUT_TYPE = INPUT_ELEMENT.type;
    expect(RENDERED_INPUT_TYPE).toBe(TYPE_PROP);

    const RENDERED_NAME_PROP = INPUT_ELEMENT.name;
    expect(RENDERED_NAME_PROP).toBe(NAME_PROP);

    const RENDERED_VALUE_PROP = INPUT_ELEMENT.value;
    expect(RENDERED_VALUE_PROP).toBe(VALUE_PROP);

    const RENDERED_REQUIRED_PROP = INPUT_ELEMENT.required;
    expect(RENDERED_REQUIRED_PROP).toBe(REQUIRED_PROP);
});
