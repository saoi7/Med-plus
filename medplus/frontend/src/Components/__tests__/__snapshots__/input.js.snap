// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check no unintended changes have been made to rendered component 1`] = `
<div
  className="add-med-input font-small undefined"
>
  <label
    className="left-padding"
  >
    undefined: 
  </label>
  <input
    className="hidden-text-box left-padding font-small"
  />
</div>
`;
